### PR TITLE
Make User registration in firebase

### LINKED ISSUE #45: GOSA16 - Make User registration in firebase
Connect user registration UI to firebase and also add registration through google sign in.

### PR DESCRIPTION
**Pull request by singhtaran1005**
GOSA16
ISSUE #45 

#### Describe the changes you've made
I have connected UI registration with Firebase and google sign in with Firebase as mentioned in the issue.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

#### Test plan (OPTIONAL)
A good test plan should give instructions that someone else can easily follow.
How someone can test your code?

#### Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](https://github.com/GameofSource-GFG/Web-Development/blob/main/CONTRIBUTING.md)** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: lib/screens/sign_up.dart [hunk 1]
```diff
@@ -131,10 +191,18 @@ class _SignUpState extends State<SignUp> {
...
-                        setState(() {
-                          validityEmail = isValidEmail(email.text);
-                          validityPassword = isValidPassword(password.text);
-                        });
+                        setState(
+                          () {
+                            validityEmail = isValidEmail(email.text);
+                            validityPassword = isValidPassword(password.text);
+                            registerToFb();
+                            Navigator.push(
```
Comment (VarunAthreya) [inline lib/screens/sign_up.dart]: Remove push function here

File: lib/screens/sign_up.dart [hunk 2]
```diff
@@ -1,25 +1,85 @@
...
   final email = TextEditingController();
   final password = TextEditingController();
   bool validityEmail = true;
   bool validityPassword = true;
+//email sign-up method
+  void registerToFb() {
+    firebaseAuth
+        .createUserWithEmailAndPassword(
+            email: email.text, password: password.text)
+        .then((result) {})
```
Comment (VarunAthreya) [inline lib/screens/sign_up.dart]: And then add it here in `.then()` function

File: lib/screens/sign_up.dart [hunk 3]
```diff
@@ -1,25 +1,85 @@
...
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:our_gfg/screens/upcoming_events_screen.dart';
 
 String emailIdErrorMessage = "";
 String passwordErrorMessage = "";
 
 class SignUp extends StatefulWidget {
-  static final String routeName = "/sign_up";
```
Comment (VarunAthreya) [inline lib/screens/sign_up.dart:8]: Add this again

**Review by VarunSAthreya — CHANGES_REQUESTED** [review]
There are 3 changes:
1. Change the position of the push statement.
2. Add signup routename
3. Install `fluttertoast` package. (Merge conflicts will be fixed later)

**Comment by singhtaran1005** [comment]
changes done

**Review by VarunSAthreya — APPROVED** [review]
Congratulations on the successful merge!
The next set of issues would be out soon.

**Pull request merged**